pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -107,10 +115,22 @@ func Materials(ctx context.Context, tro *objects.TaskRunObject) ([]common.Proven
 	sms := artifacts.RetrieveMaterialsFromStructuredResults(ctx, tro, artifacts.ArtifactsInputsResultName)
 	mats = append(mats, sms...)
 
-	if tro.Spec.Resources != nil {
+	// add task resources
+	mats = AddTaskResourcesToMaterials(ctx, tro, mats)
+
+	// remove duplicate materials
+	mats, err := RemoveDuplicateMaterials(mats)
+	if err != nil {
+		return mats, err
+	}
+	return mats, nil
+}
+
+func AddTaskResourcesToMaterials(ctx context.Context, tro *objects.TaskRunObject, mats []common.ProvenanceMaterial) []common.ProvenanceMaterial {
```

Comment:
Can we do this so that we directly mutate the `mats`?

```suggestion
func AddTaskResourcesToMaterials(ctx context.Context, tro *objects.TaskRunObject, mats []*common.ProvenanceMaterial) {
```